lib/custom/src/MShop/Service/Provider/Payment/Stripe.php

```diff
@@ -33,20 +111,276 @@ protected function getConfigPrefix()
+	 * @param array $params Request parameter if available
+	 * @return \Aimeos\MShop\Common\Item\Helper\Form\Standard Form object with URL, action and parameters to redirect to
+	 *    (e.g. to an external server of the payment provider or to a local success page)
+	 */
+	public function process(\Aimeos\MShop\Order\Item\Iface $order, array $params = [])
+	{
+		if( !isset( $params['paymenttoken'] ) ) {
+			return $this->getPaymentForm( $order, $params );
+		}
+		return $this->processOrder($order, $params);
+	}
+
+
+	/**
+	 * Returns the payment form for entering payment details at the shop site.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Iface $order Order object
+	 * @param array $params Request parameter if available
+	 * @return \Aimeos\MShop\Common\Item\Helper\Form\Iface Form helper object
+	 */
+	protected function getPaymentForm(\Aimeos\MShop\Order\Item\Iface $order, array $params)
+	{
+		$list = [];
+		$feConfig = $this->feConfig;
+
+		foreach ($feConfig as $key => $config) {
+			$list[$key] = new \Aimeos\MW\Criteria\Attribute\Standard($config);
+		}
+		$url = $this->getConfigValue(array('payment.url-self'));
+		return new \Aimeos\MShop\Common\Item\Helper\Form\Standard($url, 'POST', $list, false, $this->getHTMLform());
+	}
+
+
+	/**
+	 * Returns the configuration attribute definitions of the provider to generate a list of available fields and
+	 * rules for the value of each field in the administration interface.
 	 *
-	 * @param string $key Configuration key name
-	 * @param mixed $default Default value if no configuration is found
-	 * @return mixed Configuration value
+	 * @return array List of attribute definitions implementing \Aimeos\MW\Common\Critera\Attribute\Iface
 	 */
-	protected function getValue( $key, $default = null )
+	public function getConfigBE()
 	{
-		switch( $key )
+		$list = [];
+
+		$prefix = $this->getConfigPrefix();
+		$config = $this->beConfig;
+
+		if( $prefix !== 'omnipay' ) {
+			unset( $config['type'], $config['onsite'] );
+		}
+
+		foreach( $config as $key => $config )
 		{
-			case 'type': return 'Stripe';
-			case 'onsite': return true;
+			$config['code'] = $prefix . '.' . $config['code'];
+			$list[$prefix.'.'.$key] = new \Aimeos\MW\Criteria\Attribute\Standard( $config );
 		}
 
-		return parent::getValue( $key, $default );
+		return $list;
 	}
+
+
+	/**
+	 * Checks the backend configuration attributes for validity.
+	 *
+	 * @param array $attributes Attributes added by the shop owner in the administraton interface
+	 * @return array An array with the attribute keys as key and an error message as values for all attributes that are
+	 * 	known by the provider but aren't valid
+	 */
+	public function checkConfigBE( array $attributes )
+	{
+		$errors = parent::checkConfigBE( $attributes );
+
+		$prefix = $this->getConfigPrefix();
+		$config = $this->beConfig;
+		$list = [];
+
+		if( $prefix !== 'omnipay' ) {
+			unset( $config['type'], $config['onsite'] );
```

Comment:
Seems they are gone, at least they are not listed in the backend config any more